### PR TITLE
Feature/335 bulk import 2

### LINKED ISSUE #335: Bulk import
Bulk import would expect a zip file with individual json files for each workflow. The create workflow screen would be skipped and each workflow will be directly created in deactivated mode.

The action should complete fully or not at all. If there are validation errors, the error message should map validation errors to the workflow name that causes them. In case of a database errors at the bulk insert, it is probably impossible to map errors to specific workflows.

### PR DESCRIPTION
**Pull request by kevinwallimann**
Closes #335

### CODE REVIEW

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala [hunk 1]
```diff
@@ -153,7 +157,56 @@ class WorkflowController @Inject()(workflowService: WorkflowService) {
   @PostMapping(path = Array("/workflow/import"))
   def importWorkflow(@RequestPart("file") file: MultipartFile): CompletableFuture[WorkflowJoined] = {
     val workflowImport = ObjectMapperSingleton.getObjectMapper.readValue(file.getBytes, classOf[WorkflowImportExportWrapper])
-    workflowService.importWorkflow(workflowImport).toJava.toCompletableFuture
+    workflowService.convertToWorkflowJoined(workflowImport).toJava.toCompletableFuture
+  }
+
+  @PostMapping(path = Array("/workflows/import"))
+  def importWorkflows(@RequestPart("file") file: MultipartFile): CompletableFuture[Seq[Project]] = {
```
Comment (jozefbakus) [inline src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala:164]: Returns 200 when incorrect file is posted

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala [hunk 2]
```diff
@@ -153,7 +157,56 @@ class WorkflowController @Inject()(workflowService: WorkflowService) {
...
+    workflowService.importWorkflows(workflowImports.map(_._2.get)).toJava.toCompletableFuture
+  }
+
+  private def extractZipEntries(zipByteArray: Array[Byte]) = {
+    val bais = new ByteArrayInputStream(zipByteArray)
+    val zis = new ZipInputStream(bais)
+    val zipEntries = ArrayBuffer[(String, Array[Byte])]()
+    var entry = zis.getNextEntry
+    while (entry != null) {
+      if (!entry.getName.startsWith("__MACOSX/")) {
```
Comment (jozefbakus) [inline src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala]: Could you please explain this line ?

Comment (kevinwallimann) [reply src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala]: If you create a zip file manually on mac, it creates a hidden folder called __MACOSX. The content of that folder should be ignored.

Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala]: Cool

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala (hunk 1 above)
Comment (kevinwallimann) [reply src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala:164]: Fixed

**Review by jozefbakus — APPROVED** [review]

**Pull request merged**